x/ibc/03-connection/types/connection.go

```diff
@@ -72,6 +83,12 @@ func (c ConnectionEnd) ValidateBasic() error {
 			return sdkerrors.Wrap(ibctypes.ErrInvalidVersion, "version can't be blank")
 		}
 	}
+	if c.ID == c.Counterparty.ConnectionID {
+		return fmt.Errorf("connection ID (%s) must be different from counterparty ID (%s)", c.ID, c.Counterparty.ConnectionID)
+	}
+	if c.ClientID == c.Counterparty.ClientID {
```

Comment:
Why? It's fine if the identifiers are the same.

Reply:
why would one create a counterparty with the same client and connection ids?

Reply:
maybe they both call it "fedekunzeconnection"

Reply:
it is not likely to be common, but it's also valid from the protocol's perspective